### PR TITLE
support building pyston on ARM64 without JIT

### PR DESCRIPTION
**Pull request by undingen**
initial JIT support will come in the next week or so.

### CODE REVIEW

File: Objects/stringlib/eq.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /* Return 1 if two unicode objects are equal, 0 if not.
  * unicode_eq() is called when the hash of two unicode objects is equal.
  */
-Py_LOCAL_INLINE(int)
+__attribute__((always_inline)) Py_LOCAL_INLINE(int)
```
Comment (kmod) [inline Objects/stringlib/eq.h:6]: Just curious, why do we need this?

File: pyston/nitrous/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ferror-limit=5 -fcolor-diagnostics")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ferror-limit=5 -fcolor-diagnostics")
 
-set(LLVM_LIB_DEPS LLVMCore LLVMSupport  LLVMBitReader LLVMAsmParser  LLVMTransformUtils LLVMScalarOpts  LLVMOrcJIT LLVMX86CodeGen LLVMX86AsmParser)
+set(LLVM_LIB_DEPS LLVMCore LLVMSupport  LLVMBitReader LLVMAsmParser  LLVMTransformUtils LLVMScalarOpts  LLVMOrcJIT LLVMAArch64CodeGen LLVMAArch64AsmParser)
```
Comment (kmod) [inline pyston/nitrous/CMakeLists.txt]: Oh hmm, how is this passing the x86 github CI? Do we not use codegen anymore?

**Review by kmod — APPROVED** [review]
lgtm except I'm confused by how it works to always use the aarch64 libraries even on x86

File: pyston/nitrous/CMakeLists.txt (hunk 2 above)
Comment (undingen) [reply pyston/nitrous/CMakeLists.txt]: I took this commit from your arm branch. I does not work on x86 but I did add the arm/x86 check in my [3c623f5](https://github.com/pyston/pyston/pull/176/commits/3c623f5b27fa14ae1a5a30ce288a180096d40edf) commit.
(gettiing `undefined symbol: LLVMInitializeX86TargetInfo` without my fix)

Comment (undingen) [reply pyston/nitrous/CMakeLists.txt]: image

**Comment by kmod** [comment]
ahhh makes sense

File: Objects/stringlib/eq.h (hunk 1 above)
Comment (undingen) [reply Objects/stringlib/eq.h:6]: I just rechecked this and I'm getting this error without it:
```
/usr/bin/ld: aot_optimized.o: in function `lookdict_unicode_nodummy':
/home/undingen/dev/pyston/build/bc_build/../../Objects/dictobject.c:892: undefined reference to `unicode_eq'
```
Seems like on ARM the compiler makes a bit different inlining decision.

Comment (undingen) [reply Objects/stringlib/eq.h:6]: And the problem is that this function is defined in a header which get included from `dictobject.c` and `unicodeobject.c` So just removing the static will make it complain about having it defined twice. (`Py_LOCAL_INLINE` is `static inline`)

Comment (kmod) [reply Objects/stringlib/eq.h:6]: Ah I see, yeah I think removing `static` has been our more typical approach so I wasn't sure why that wasn't used here, but now I get it

**Pull request merged**